mlir/lib/Dialect/Linalg/IR/LinalgOps.cpp

```diff
@@ -5038,6 +5126,9 @@ struct FoldTensorCastPackOp : public OpRewritePattern<PackOp> {
     if (!tensor::hasFoldableTensorCastOperand(op))
       return failure();
 
+    if (!op.hasPureTensorSemantics())
```

Comment:
Please add a TODO for memref. It is valid to fold memref.cast away as well.

Reply:
Done